enumerables.rb

```diff
@@ -0,0 +1,171 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+#  rubocop:disable Metrics/MethodLength,Style/CaseEquality
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |item|
+      yield item
+    end
+
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    each do |item|
+      yield item, index
+      index += 1
+    end
+
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    results_array = []
+    each do |item|
+      next unless yield item
+
+      results_array << item
+    end
+    results_array
+  end
+
+  def my_all?(arg = nil)
+    result = true
+    if block_given?
+      each do |item|
+        result = false unless yield item
+      end
+    elsif arg
+      if arg.is_a?(Regexp)
+        each do |item|
+          result = false unless arg.match(item)
+        end
+      else
+        each do |item|
+          result = false unless arg === item
+        end
+      end
+    else
+      each do |item|
+        result = false unless item
+      end
+    end
+    result
+  end
+
+  def my_any?(arg = nil)
+    result = false
+    if block_given?
+      each do |i|
+        result = true if yield i
+
+        next
+      end
+    elsif arg
+      if arg.is_a?(Regexp)
+        each do |item|
+          result = true if arg.match(item)
+        end
+      else
+        each do |item|
+          result = true if arg === item
+        end
+      end
+    else
+      each do |i|
+        result = true if i
+
+        next
+      end
+    end
+    result
+  end
+
+  def my_none?(arg = nil)
+    result = true
+    if block_given?
+      each do |item|
+        result = false if yield item
+      end
+    elsif arg
+      if arg.is_a?(Regexp)
+        each do |item|
+          result = false if arg.match(item)
+        end
+      else
+        each do |item|
+          result = false if arg == item
+        end
+      end
+    else
+      each do |item|
+        result = false if item
+      end
+    end
+    result
+  end
+
+  def my_count(number = nil)
+    input = Array(self)
+
+    count = 0
+    if number
+      each do |i|
+        count += number if i == number
+      end
+    end
+
+    return input.length unless block_given?
+
+    each do |i|
+      count += 1 if yield i
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    end_array = []
+    item = nil
+    each do |i|
+      if proc
+        item = proc.call(i)
+      else
+        return to_enum(:my_map) unless block_given?
+
+        item = yield i
+      end
+      end_array.push item
+    end
+    end_array
+  end
+
+  def my_inject(acc = 0, operation = nil)
+    acc = Array(self)[0] if acc.is_a?(Symbol)
+
+    each do |i|
+      acc = yield(i, acc) if block_given?
+
+      acc = i.send(operation, acc) unless operation.nil?
+    end
+    acc
+  end
+end
```

Comment:
# my_inject
- [ ] Raises a `LocalJumpError` when no block or argument is given.
`Hint` => `expect { array.my_inject }.to raise_error(LocalJumpError)`
       expected LocalJumpError but nothing was raised.

- [ ] The method should also search for the longest word in an array of strings.

- [ ] When a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range

- [ ] When a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method.
- [ ] When a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range